azure/ConsiderationsForServiceDesign.md

```diff
@@ -0,0 +1,99 @@
+
+### Focus on Hero Scenarios
+It is important to realize that writing an API is, in many cases, the easiest part of providing a delightful developer experience. There are a large number of downstream activities for each API, e.g. testing, documentation, client libraries, examples, blog posts, videos, and supporting customers in perpetuity. In fact, implementing an API is of miniscule cost compared to all the other downstream activities. 
+
+*For this reason, it is <b>much better</b> to ship with fewer features and only add new features over time as required by customers.*
+
+Focusing on hero scenarios reduces development, support, and maintenance costs; enables teams to align and reach consensus faster; and accelerates the time to delivery. A telltale sign of a service that has not focused on hero scenarios is "API drift," where endpoints are inconsistent, incomplete, or juxtaposed to one another. 
+
+> :white_check_mark: **DO** define "hero scenarios" first including abstractions, naming, relationships, and then define the API describing the operations required
+> 
+> :white_check_mark: **DO** provide example code demonstrating the "Hero Scenarios"
+> 
+> :no_entry: **DO NOT** proactively add APIs for speculative features customers might want 
+
+> :white_check_mark: **DO** consider how your abstractions will be represented in different high-level languages.
+
+> :white_check_mark: **DO** develop code examples in at least one dynamically typed language (for example, Python or JavaScript) and one statically typed language (for example, Java or C#) to illustrate your abstractions and high-level language representations.
+
+### Start with your API Definition
+Understanding how your service is used and defining its model and interaction patterns--its API--should be one of the earliest activities a service team undertakes. It reflects the abstractions & naming decisions and makes it easy for developers to implement the hero scenarios.  
+
+> :white_check_mark: **DO** provide an [OpenAPI Definition](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/2.0.md) (with [autorest extensions](https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md)) describing the service. The OpenAPI definition is a key element of the Azure SDK plan and is essential for documentation, usability and discoverability of services.
+
+### Use Previews to Iterate 
+ Before releasing your API plan to invest significant design effort, get customer feedback, & iterate through multiple preview releases. This is especially important for V1 as it establishes the abstractions and patterns that developers will use to interact with your service.
+
+> :ballot_box_with_check: **YOU SHOULD**  write and test hypotheses about how your customers will use the API.
+> 
+> :ballot_box_with_check: **YOU SHOULD**  release and evaluate a minimum of 2 preview versions prior to the first GA release.
```

Comment:
This whole section does not make it clear how we think customers would evaluate the APIs. Do we want customers to call REST API directly? Or do we need libraries. If the later, are libraries a good way to evaluate REST APIs?

Reply:
Our APIs are consumed in so many different ways that I don't think we should, in this document, recommend or prescribe how an preview should be conducted.